### DISCUSSION TITLE
Trouble Binding to Shell Script; Shell Script invokes LXD command

### DISCUSSION THREAD

**Discussion by Mnchstrcityblues79** [i3/i3#4721] [Q&A] [answered]
Good morning.  I had previous posted this at reddit but then noticed y'all moved over to here.  My problem seems relatively simple: I can't get a bindsym to work.  Here is my original post: 

> Hey all, I'm pulling out my hair, here.
> I want to bind "mod+b" to lxdfox.sh
> i3 config (in pertinent part) is:
> bindsym $mod+b exec /usr/local/binlxdfox.sh
> 

> 
> 
> lxdfox.sh is as follows:```
> 
> #!/bin/bash
> 
> lxc exec firefox -- sudo --user ubuntu --login ./firefox_audio.sh```
> 
> File access is:
> -rwxr-xr-x  1 user user   79 Dec 10 23:04 lxdfox.sh
> 
> So I mod the config, i3-msg restart, press mod+b and ... *crickets*
> Any ideas?  Thank you!

Note: invoking `lxc exec firefox -- sudo --user ubuntu --login ./firefox_audio.sh` works perfectly.

The dump log (starting with invoking the bindsym mod+b) is here: https://pastebin.com/N8pguFqk

Any ideas?  Thank you!

**Comment by orestisfl** [comment, marked as answer]
Per the log, i3 execs the script successfully. Does your script depend on environmental variables that might not be set for i3? If yes, you will have to export them before executing i3.

**Reply by Mnchstrcityblues79** [reply]
Thank you for your response!  Unfortunately I know just enough to get in trouble and not enough to answer your question.  When you say environmental variables, how do you mean?The LXC command itself does pass both DISPLAY and PULSE_AUDIO to the container, but would i3 necessarily need to know that (if that makes sense).

**Reply by orestisfl** [reply]
lxc inherits the environmental variables that its parent has. In this case, i3.

When in the terminal, the parent is your $SHELL.

You can easily compare the two environments by binding `bindsym $mod+backslash exec env > /tmp/a` and then running `env > /tmp/b` from your terminal. Then run `diff -u /tmp/a /tmp/b` and see what's missing.

**Reply by Mnchstrcityblues79** [reply]
I marked your response as answer because it actually led me down the correct path!

Turns out, i3 did not know where to find the command "lxc" which was within the .sh.  So, I placed the full path of the command "lxc" in the config and *boom* success!  Thank you!
